ros/src/tl_detector/tl_detector.py

```diff
@@ -101,7 +109,14 @@ def get_closest_waypoint(self, pose):
 
         """
         #TODO implement
-        return 0
+        index = -1
+        dist_min = float("inf")
+        for i, wp in enumerate(self.waypoints):
+            dist = self.calculate_distance(pose.position.x, wp.pose.pose.position.x, pose.position.y, wp.pose.pose.position.y)
+            if dist < dist_min:
+                dist_min = dist
+                index = i
+        return index
```

Comment:
Here we can do a minor change to limit how much iterations the function is doing. The waypoints are ordered sequentially, which means the distance will decrease until it finds the closest point, then it will start to grow again, until the end of the loop. We can add an else to the if inside the for so it breaks the first time it finds a point that it has a bigger distance. 

`
if dist < dist_min:
  dist_min = dist
   index = i
else:
  break
`

Reply:
Got it, how can I missed that, I will make the change soon.

Reply:
Hi Carlos, I made the change as well as changed it to 20Hz loop. Let me know if you have more concerns.

Reply:
Thank you Jun